core/arch/asm_defines.asm

```diff
@@ -99,23 +99,71 @@
 # define START_DATA .data
 # define START_FILE .text
 # define END_FILE /* nothing */
+
+#if defined(MACOS) && defined(AARCH64)
```

Comment:
Unfortunately we don't have clang-format on this file (not sure why not?  It's all cpp despite the name .asm; should try to add) which would auto-indent for us.  This should be indented and the ones below as well.